table/block_fetcher.h

```diff
@@ -45,7 +45,7 @@ class BlockFetcher {
   RandomAccessFileReader* file_;
   FilePrefetchBuffer* prefetch_buffer_;
   const Footer& footer_;
-  const ReadOptions& read_options_;
+  const ReadOptions read_options_;
```

Comment:
Looks ok to me, but ... 
Wouldn't it be better to pass in a variable that outlasts the scope at the BlockFetcher constructor call-sites? That could help us avoid invoking a copy constructor for ReadOptions when constructing BlockFetcher.

Reply:
The ReadOptions struct is not big so I expect the overhead of copying would not be a big deal. It sounds painful if each time we instantiate BlockFetcher we need to remember to create a ReadOptions first. What do you think?